rasterio/warp.py

```diff
@@ -241,6 +244,14 @@ def reproject(
         **kwargs)
 
 
+def res_tuple(res):
+    try:
```

Comment:
I concede - I found a few posts that back up your claim of being more pythonic.  Not that I'm a fan of checking `isinstance` - really we should be checking to see if something can do something rather than 'is' something; which I guess we get by way of exception handling here.  One of those problems of languages that let you do the same thing a few different ways :smile: .  

I guess my thinking of inlining this was that it appeared specific to handling arguments of that function, and didn't seem designed for larger use.  Seems like many times when we drop in a function at the top level in one of the python modules, we get a question about whether that is part of the public API.  Was trying to avoid that.